tests/test_plugin_loader.py

```
@@ -2310,6 +2310,435 @@ def test_clone_or_update_repo_checkout_fails_fallback(
+    def test_clone_or_update_repo_validated_handles_clone_exception(
+        self, _mock_isdir, mock_run_git, mock_logger
+    ):
+        """Validated clone path should catch and log raised git errors."""
+        mock_run_git.side_effect = FileNotFoundError("git")
+
+        result = pl._clone_or_update_repo_validated(
+            "https://github.com/user/repo.git",
+            "commit",
+            "deadbeef",
+            "repo",
+            "/tmp/plugins",
```

Ruff (0.15.7)

[error] Probable insecure usage of temporary file or directory: "/tmp/plugins"

(S108)